### PR TITLE
Refactored .refined_sampling, removed extraction of dim names

### PR DESCRIPTION
**Pull request by emdann**

### CODE REVIEW

**Comment by MikeDMorgan** [comment]
Can I just double-check there is a definite speed improvement, and that identical results are generated with the previous version?

**Comment by emdann** [comment]
sure, go ahead

**Comment by MikeDMorgan** [comment]
> sure, go ahead

I mean, have you profiled to make sure it's faster and checked that the output is identical to previously?

**Comment by emdann** [comment]
Ah sorry LOL, I've tested on a small dataset, let me try again on a bigger one.

**Comment by MikeDMorgan** [comment]
Cool, thanks!

**Comment by emdann** [comment]
On Milo object with 10000 cells 

Old `makeNhoods`:
```r
> system.time(milo2 <- makeNhoods(milo, d=11))
Checking valid object
   user  system elapsed 
  3.612   0.008   3.618 
```

New `makeNhoods`:
```r
> system.time(milo1 <- makeNhoods(milo, d=11))
Checking valid object
   user  system elapsed 
  1.136   0.004   1.140 
```

The sampled cells are the same
```
> all(names(nhoods(milo1))==names(nhoods(milo2)))
[1] TRUE
```

**Comment by MikeDMorgan** [comment]
> On Milo object with 10000 cells
> 
> Old `makeNhoods`:
> 
> ```r
> > system.time(milo2 <- makeNhoods(milo, d=11))
> Checking valid object
>    user  system elapsed 
>   3.612   0.008   3.618 
> ```
> 
> New `makeNhoods`:
> 
> ```r
> > system.time(milo1 <- makeNhoods(milo, d=11))
> Checking valid object
>    user  system elapsed 
>   1.136   0.004   1.140 
> ```
> 
> The sampled cells are the same
> 
> ```
> > all(names(nhoods(milo1))==names(nhoods(milo2)))
> [1] TRUE
> ```

Nice!
